File: solutions/javascript/lasagna-master/1/lasagna-master.js
/// <reference path="./global.d.ts" />
// @ts-check

/**
 * Implement the functions needed to solve the exercise here.
 * Do not forget to export them so they are available for the
 * tests. Here an example of the syntax as reminder:
 *
 * export function yourFunction(...) {
 *   ...
 * }
 */
export function cookingStatus(remainingTime){
  if(remainingTime === 0) return "Lasagna is done."
  if(remainingTime) return "Not done, please wait."
  return "You forgot to set the timer."
}

export function preparationTime(layers, timeInMinutes = 2){
  return layers.length * timeInMinutes
}

export function quantities(items){
  let noodleQty = 0, sauceQty = 0;

  for(let item of items) {
      if(item === "sauce") sauceQty += 0.2
      if(item === "noodles") noodleQty += 50
  }

  return {noodles: noodleQty, sauce: sauceQty}
}

export function addSecretIngredient(friendList, myList){
  const [lastItem] = friendList.slice(-1)
  myList.push(lastItem)
}

export function scaleRecipe(recipeItems, portionToCook = 1){
  let scaledQty = {}
   for(const [key, value] of Object.entries(recipeItems)){
     scaledQty[key] = (value/2)*portionToCook
   }
  return scaledQty
}

File: solutions/go/lasagna-master/2/lasagna_master.go
package lasagnamaster

// TODO: define the 'PreparationTime()' function
func PreparationTime(layers []string, avgTimePerLayer int) int{
	if avgTimePerLayer == 0 {
		avgTimePerLayer = 2
	}
	return len(layers) * avgTimePerLayer
}
// TODO: define the 'Quantities()' function
func Quantities() int {
	return 0
}

// TODO: define the 'AddSecretIngredient()' function
func AddSecretIngredient() string{
	return "s"
}

// TODO: define the 'ScaleRecipe()' function
func ScaleRecipe() int {
	return 0
}